src/components/FooterFilter.tsx

```diff
@@ -0,0 +1,58 @@
+import cn from 'classnames';
+import { useTodoContext } from '../context/TodoContext';
+import { Filter } from '../types/Filter';
+
+export const FooterFilter: React.FC = () => {
+  const { todos, filter, setFilter } = useTodoContext();
+
+  return (
+    <>
+      <nav className="filter">
+        <a
+          href="#/"
+          className={cn('filter__link', {
+            selected: filter === Filter.ALL,
+          })}
+          onClick={(event) => {
+            event.preventDefault();
+            setFilter(Filter.ALL);
+          }}
+        >
+          All
+        </a>
+
+        <a
+          href="#/active"
+          className={cn('filter__link', {
+            selected: filter === Filter.ACTIVE,
+          })}
+          onClick={(event) => {
+            event.preventDefault();
+            setFilter(Filter.ACTIVE);
+          }}
+        >
+          Active
+        </a>
+
+        <a
+          href="#/completed"
+          className={cn('filter__link', {
+            selected: filter === Filter.COMPLETED,
+          })}
+          onClick={(event) => {
+            event.preventDefault();
+            setFilter(Filter.COMPLETED);
+          }}
+        >
+          Completed
+        </a>
+      </nav>
+
+      {todos.some((todo) => todo.completed) && (
```

Comment:
It will be clearer and better to store the result of this check in a variable and use the variable here.